sql/store/commitment.go

```diff
@@ -0,0 +1,52 @@
+package store
+
+import (
+	"fmt"
+
+	"github.com/spacemeshos/go-spacemesh/common/types"
+	"github.com/spacemeshos/go-spacemesh/sql"
+)
+
+const CommitmentATXKey = "CommitmentATX"
+
+// AddCommitmentATX adds the id for the commitment atx to the key-value store.
+func AddCommitmentATX(db sql.Executor, atx types.ATXID) error {
```

Comment:
why does it need to be stored like this? as you mentioned commitment atx is referenced in the nipost challenge in the first atx.

please remove this file

Reply:
The reason it is stored like this is because a) its simpler and faster to look it up than having to look up the first ATX of the node itself and get it from there and b) the commitment ATX has to be determined before the first ATX is published.

It is similar to the `NiPoSTChallenge`, that is also handled by this package. It needs to be persisted before the ATX using it is published because the node could shut down in between and then needs to restore the same value to continue work where it left off.